### PR TITLE
Remove RFXCom specifics from CDomoticzHardwareBase

### PR DESCRIPTION
**Pull request by emontnemery**
Move RFXCom specifics from CDomoticzHardwareBase to CRFXBase, get rid of readQueueMutex from most HW.

### CODE REVIEW

**Comment by emontnemery** [comment]
@gizmocuz I would appreciate if you could help verify that DomoticzTCP is working as expected with this change.

**Comment by gizmocuz** [comment]
Why not remove the whole readQueueMutex ? But i think this was needed when the hardware class was terminating/closing
DomoticzTCP is used in the client/server setup. For instance, you have a main domoticz, install another instance of domoticz, let it connect to your main. (with a specific user with viewer rights and maybe some specific sensors or all)

**Comment by emontnemery** [comment]
The readQueueMutex is still needed to ensure modifications to the readbuffer are done in a safe way, for example these lines in mainworker:
```
CRFXBase *pRFXBase = reinterpret_cast<CRFXBase *>(pHardware);
 pRFXBase->SendResetCommand();
```

OK, I'll give client/server setup a try.

File: hardware/OpenWebNetTCP.cpp [hunk 1]
```diff
@@ -419,6 +419,7 @@ void COpenWebNetTCP::MonitorFrames()
 				}
 				else
 				{
+					//TODO: This mutex is only needed if it's not safe to call ParseData concurrently, is that the case?
```
Comment (gizmocuz) [inline hardware/OpenWebNetTCP.cpp]: Yes this seems possible from the senddata function

**Comment by emontnemery** [comment]
@gizmocuz DomoticzTCP is working well. I rebased and fixed review comments, should be ready for merge!

**Comment by gizmocuz** [comment]
Thanks ! But the proxyclient as well as DomoticzTCP should send valid messages right ? Why the bypass of the correctness check ? This should be done

**Comment by emontnemery** [comment]
@gizmocuz Sure, proxyclient and DomoticzTCP should send valid messages, it's just that there's no guarantee they're RFXCom messages.
CheckValidRFXData() checks that the message is an RFXCom message, so calling CheckValidRFXData() doesn't make sense.

**Comment by gizmocuz** [comment]
Thanks!!

**Comment by gizmocuz** [comment]
Your comment is right, but as soon as we are going to go for a json serialization  approach everything has to go via a big checking/de-serializing routine

**Comment by gysmo38** [comment]
Hello I use domoticz and rfxcom. Since last update if i send lighting4 with rfxcom it stop working. On rfxcom i get a red led stuck. I need to unplug and plug the rfxcom to get it work again.

This commit could be my problem?

**Comment by emontnemery** [comment]
Can you open an issue for that problem please?

**Comment by gizmocuz** [comment]
That could be an endless loop in sending commands... do you have other switches that you control (non-lighting 4, but maybe lighting 1 or 2) ... do they still work correctly ?

**Comment by gysmo38** [comment]
yes i have other switch that use blyss and somfy. It work well if I not use lighting4. If I use one lighting4 switch it stop working for all rfxcom devices.

**Pull request merged**